tests/conftest.py

```diff
@@ -1,16 +1,114 @@
 import pytest
-from functools import wraps
 
-from ethereum.tools import tester
+# from functools import wraps
+import eth_tester
+import web3
+
+from eth_tester import (
+    EthereumTester,
+)
+from eth_tester.exceptions import (
+    TransactionFailed
+)
+from web3.providers.eth_tester import (
+    EthereumTesterProvider,
+)
+
+from web3 import (
+    Web3,
+)
+from web3.contract import (
+    ConciseContract,
+)
+from web3.exceptions import (
+    ValidationError
+)
 from vyper.parser.parser_utils import (
     LLLnode
 )
 from vyper import (
     compile_lll,
-    optimizer,
     compiler,
+    optimizer,
+)
+from vyper.utils import (
+    sha3
 )
-from ethereum import utils as ethereum_utils
+
+############
+# PATCHING #
+############
+
+
+def Filter_remove(self, *values):
+
+    def get_key(v):
+        return v.get('transaction_hash'), v.get('log_index'), v.get('transaction_index')
+
+    values_to_remove = set([
+        get_key(value)
+        for value in values
+    ])
+
+    queued_values = self.get_changes()
+    self.values = [
+        value
+        for value
+        in self.get_all()
+        if get_key(value) not in values_to_remove
+    ]
+    for value in queued_values:
+        if get_key(value) in values_to_remove:
+            continue
+        self.queue.put_nowait(value)
+eth_tester.utils.filters.Filter.remove = Filter_remove
```

Comment:
This looks like a good case for using the [`monkeypatch`](https://docs.pytest.org/en/latest/monkeypatch.html) fixture that pytest provides assuming you are not aware of it.